components/tidb_query_datatype/src/codec/mysql/time/mod.rs

```diff
@@ -369,11 +404,100 @@ mod parser {
             buffer = rest;
         }
 
+        let mut tz_offset = 0i32;
+        let mut tz_sign: &[u8] = b"".as_ref();
```

Comment:
You can just use `let tz_sign: &[u8];` without giving an initial value. Rust compiler will check if it is assigned before using.